### ISSUE TITLE
BUG: No spatial index written to geopackage

### ISSUE THREAD

**Issue by joostmeulenbeld** [geopandas/pyogrio#187]
When writing a geopackage file using `pyogrio`, an error message from sqlite is shown that `rtree` is not installed. The GPKG file is produced, but upon inspection no spatial index is present. The files can be read in `pyogrio` or QGIS. However, the lack of spatial index is problematic for us because reading parts of a geopackage file without spatial index is much slower. Below are the error I get, a benchmark showing difference in read speed, and some notes.

## Error when writing geopackage
```python
import geopandas as gpd
import pyogrio
from shapely import Point

gdf = gpd.GeoDataFrame(geometry=[Point(0, 0)], crs="EPSG:28992")
pyogrio.write_dataframe(gdf, "geometry_file.gpkg")
```
Error message:
```
ERROR 1: sqlite3_exec(CREATE VIRTUAL TABLE "rtree_geometry_file_geom" USING rtree(id, minx, maxx, miny, maxy)) failed: no such module: rtree
```

When inspecting the resulting geopackage, indeed all the tables named `rtree_*` are not present, indicating no spatial index was written.

## Timing difference in read speed
The speed difference for the below example is a factor 4 when reading a selection of the geopackage.

Create a file with 1 million points in range (0-1, 0-1):
```python
import geopandas as gpd
import pyogrio
import random
from shapely import Point

gdf = gpd.GeoDataFrame(geometry=[Point(random.random(), random.random()) for _ in range(1_000_000], crs="EPSG:28992")
pyogrio.write_dataframe(gdf, "geometry_file.gpkg")
```
Then convert the file to a new geopackage with `ogr2ogr`. This will automatically add a spatial index:
```bash
ogr2ogr geometry_file_with_rtree.gpkg  geometry_file.gpkg
```

Timing the read speed difference with `fiona`:
```python
import timeit

time_with_spatial_index = (
    timeit.timeit(
        'gpd.read_file("geometry_file_with_rtree.gpkg", engine="fiona", bbox=(0, 0, 0.1, 0.1))',
        setup="import geopandas as gpd",
        number=10,
    )
    / 10
)
time_without_spatial_index = (
    timeit.timeit(
        'gpd.read_file("geometry_file.gpkg", engine="fiona", bbox=(0, 0, 0.1, 0.1))',
        setup="import geopandas as gpd",
        number=10,
    )
    / 10
)

print(f"Time for reading file without spatial index: {time_without_spatial_index}")
print(f"Time for reading file with spatial index: {time_with_spatial_index}")
```
Result:
```
Time for reading file without spatial index: 1.00816962569952
Time for reading file with spatial index: 0.2739204873993003
```
## Notes
* Installing the python module rtree does not change anything.
* This is probably related to https://github.com/geopandas/pyogrio/issues/179

OS: Pop!_OS 22.04 LTS (Linux)
Python version: `3.10.8`
`pyogrio` version: `0.4.2` (installed from PyPI)

**Comment by jorisvandenbossche** [comment]
@joostmeulenbeld thanks for the report!

So I suppose this is due to how we built the wheels, and that the sqlite included in the binary wheels is not built with rtree enabled. We built the dependencies using vcpkg, and checking there, it is indeed the case that until recently, the GDAL build recipe didn't mention sqlite3 + rtree as dependency, but just sqlite. 
Now, with the latest wheels built for the development version, we are now using a more recent GDAL version, and the vcpkg recipe has been updated to include rtree (https://github.com/dg0yt/vcpkg/blob/4b689a3411328414ad216c4ee949a39dfa1b6dd7/ports/gdal/vcpkg.json#L172-L175).

So it _might_ the case this is already solved (and we need to make a release to get the new wheels out). 

Would you like to test this? You can install the latest wheel by going to https://github.com/geopandas/pyogrio/actions/runs/3793070654, and at the bottom of that page download the "artifact". Unpack this archive, and then `pip install path/to/archive/pyogrio-0.4.2+12.g40e7659-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`

**Comment by joostmeulenbeld** [comment]
@jorisvandenbossche Thanks for the very quick reaction in this festive period :fireworks: 

I tested with the wheels, and indeed the problems I saw are no longer present!
* The created geopackages have a spatial index
* bbox-reading them with `fiona` engine is now fast as well
* There are also no longer problems with reading geopackages with a spatial index in them using `pyogrio`, which I used to have issues with as well

> and we need to make a release to get the new wheels out

Do you have any idea when we might expect a such a release?

Thanks a lot for all the effort, geopandas and pyogrio are amazing packages!

**Comment by jorisvandenbossche** [comment]
@joostmeulenbeld thanks for testing and verifying it now works!

> Do you have any idea when we might expect a such a release?

I want to finalize the currently open PRs, and then planning to do a release, hopefully somewhere next week.

**Comment by jorisvandenbossche** [comment]
I opened https://github.com/geopandas/pyogrio/issues/192 to track the upcoming release.

**Comment by joostmeulenbeld** [comment]
I confirmed that this issue is now also fixed in the `0.5.0` version released on PyPI. Thanks!
